lib/mqtt_abstraction_impl.cpp

```diff
@@ -484,6 +472,7 @@ void MQTTAbstractionImpl::unregister_handler(const std::string& topic, const Tok
             EVLOG_verbose << fmt::format("Unsubscribing from {}", topic);
             this->unsubscribe(topic);
         }
+        this->message_handlers.erase(topic);
```

Comment:
Opened a new PR for this change which helps reduce the amount of idle threads quite significantly: https://github.com/EVerest/everest-framework/pull/229
In general this design needs to be refactored, but that's probably best done in a future PR